### PR TITLE
560 Update the Static Site Navigation

### PR DESCRIPTION
**Pull request by johnfrenchxyz**
So, I think this may be ready to deploy to test, along with the update to the CSS Repo. This will give me some time to look at it in a real environment, on various phones, etc.

List of changes/tweaks:
- HTML changes are pretty self-explanatory, but I consolidated where I could, cleaned up the logo SVG, added comments, etc.
- Implemented [Feather Icons](https://feathericons.com). They are open source and super easy and fun to work with. I used them in the design approval, so I needed them here. Can slowly phase out other icons in the future to reduce load.
- Wrote nav logic in vanilla JS to eventually reduce our dependence on jQuery, which we could remove in the future for a potential performance win/stay up with the times.

### CODE REVIEW

**Comment by karlmdavis** [comment]
FYI: The CMS employee web filtering/firewall blocks the feathericons.com site -- it has it listed as "shareware-and-freeware". Can you file a JIRA ticket to have someone with access to the CMS internal network (e.g. Mark or myself) go fight with the internal IT folks to get that fixed?

(And yes, I'm horrified that this is a thing, too.)

**Comment by johnfrenchxyz** [comment]
Sure thing @karlmdavis - I can make that ticket. Any idea of who that should go to (new to the CMS structure)?

**Comment by karlmdavis** [comment]
Me, I suppose.

File: assets/js/main.js [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
+
+// Define the mobile nav trigger button - this opens and closes the mobile nav
+const mobileNavTriggerButton = document.querySelector('.mobile-nav-trigger-button');
+
+// Add the click ation to the mobile nav trigger
+mobileNavTriggerButton.addEventListener('click', () => {
+  mobileNavContainer.classList.toggle('is-visible');
+  mobileNavTriggerButton.classList.toggle('trigger-active');
+  // Define the mobile nav icon
+  let mobileNavIcon = document.querySelector('.mobile-nav-icon');
```
[the remark this answers is not in the document]
Comment (johnfrenchxyz) [reply assets/js/main.js]: I definitely agree with that thinking. I went ahead and moved the `mobileNavTriggerText` to be global with the other global vars and will commit in just a bit. But for some reason, if I make the `mobileNavIcon` global, it is not triggering the feather `toSVG()` as expected. I am investigating.

[the remark this answers is not in the document]
Comment (johnfrenchxyz) [reply assets/js/main.js]: Ah - I figured it out. It was related to script order. I moved my custom JS to be after the Feather Icons JS and it seems to be working fine with that icon defined globally now.

[the remark this answers is not in the document]
Comment (johnfrenchxyz) [reply assets/js/main.js]: I am going to resolve this for now. Take a look at the new commit in just a minute and let me know if you have other thoughts.

File: assets/js/main.js [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
+
+  // Define the text for the mobile nav trigger button
+  let mobileNavTriggerText = document.querySelector('.mobile-nav-trigger-text');
+
+  // Set up toggle for the menu icon and text
+  if (mobileNavContainer.classList.contains('is-visible') === true) {
+    mobileNavTriggerText.innerHTML = 'Close';
+    mobileNavIcon.innerHTML = feather.icons['x'].toSvg();
+    // Add an event listenter to close the nav if the window is resized
+    window.addEventListener('resize', () => {
```
[the remark this answers is not in the document]
Comment (johnfrenchxyz) [reply assets/js/main.js]: @whytheplatypus I totally understand your point and good catch on it continuing to fire. I totally missed that. I made the function inside the event listener global, and then limited the number of times the resize check would fire with a `for` loop, then I remove the listener. I definitely don't think this is the cleanest way to do this, but it at least solves the problem for now.

File: assets/js/main.js [hunk 3]
```diff
@@ -13,28 +13,35 @@ mobileNavContainer.innerHTML += mobileNavContent;
...
     mobileNavIcon.innerHTML = feather.icons['x'].toSvg();
     // Add an event listenter to close the nav if the window is resized
-    window.addEventListener('resize', () => {
-      // Close and reset the nav
-      mobileNavTriggerText.innerHTML = 'Menu'
-      mobileNavIcon.innerHTML = feather.icons['menu'].toSvg();
-      mobileNavContainer.classList.remove('is-visible');
-      mobileNavTriggerButton.classList.remove('trigger-active');
-    });
+    window.addEventListener('resize', closeNavOnResize);
```
Comment (whytheplatypus) [inline assets/js/main.js]: can this be registered at the top level, and have a
```
if (mobileNavContainer.classList.contains('is-visible') === false) { return }
```
at the top of the closeNavOnResize functions?

I think that would let you get rid of the `for` loop as well

File: assets/js/main.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnfrenchxyz) [reply assets/js/main.js]: Oh, gotcha, gotcha. Good thought. I think that would be good. Will test it and show in a sec.

File: assets/js/main.js (hunk 3 above)
Comment (johnfrenchxyz) [reply assets/js/main.js]: Oh, gotcha, gotcha. Good thought. I think that would be good. Will test it and show in a sec.

Comment (johnfrenchxyz) [reply assets/js/main.js]: @whytheplatypus just added a commit. This looks like it is working now. I have to run to a lunch appointment, but let me know if there is anything else I missed here. Thanks again!

**Pull request merged**